lib/index.ts

```diff
@@ -1,147 +1,147 @@
-            // cors is exempt from authentication
-            if (req.method === 'OPTIONS') {
-                res.set('Access-Control-Allow-Origin', '*')
-                res.set('Access-Control-Allow-Headers', 'Authorization, Origin, X-Requested-With, Content-Type, Accept')
-                return res.status(200).end()
-            }
-
-            // the rest of the stack needs to authenticate
-            passport.authorize(['oauth-bearer', 'oauth-bearer-b2c'], (err, user, info) => {
-                if (!err && user) {
-                    next()
-                } else {
-                    res.status(401).end()
-                }
-            })(req, res, next)
-        })
-    }
-
-    // configure heartbeat if needed
-    if (!optIsFalsey(opts.heartbeatEnabled)) {
-        const heartbeatRouter = heartbeatRouterCreator({
-            peerList: peerList,
-            timeoutPeriod: opts.heartbeatMs || 30 * 1000,
-            gcInterval: opts.heartbeatGcMs || 15 * 1000
-        })
-
-        // use the heartbeat peerList instead
-        peerList = heartbeatRouter.peerList
-
-        app.use(heartbeatRouter)
-    }
-
-    // configure capacity if needed
-    if (!optIsFalsey(opts.capacityEnabled)) {
-        const capacityRouter = capacityRouterCreator({
-            peerList: peerList
-        })
-
-        // use the capacity peer list instead
-        peerList = capacityRouter.peerList
-
-        app.use(capacityRouter)
-    }
-
-    // configure peer recognition router if needed
-    if (!optIsFalsey(opts.recognitionEnabled)) {
-        const recognitionRouter = recognitionRouterCreator({
-            peerList: peerList
-        })
-
-        // use the recognition peer list instead
-        peerList = recognitionRouter.peerList
-
-        app.use(recognitionRouter)
-    }
-
-    // configure the webrtc-signal-http router
-    const signalRouter = signalRouterCreator({...opts,
-        // we use a predefined peerList in case we need to support heartbeats
-        peerList: peerList,
-
-        // see above, we use a different logger to capture more scenarios
-        enableLogging: false
-    })
-
-    // configure webrtc-signal-http-publisher if needed
-    if (!optIsFalsey(opts.publishState)) {
-        app._publisher = new Publisher(signalRouter)
-    }
-
-    // use webrtc-signal-http router
-    app.use(signalRouter)
-
-    // expose the peerList on the app
-    app.peerList = peerList
-
-    // return a configured app, ready to be used
-    return app
-}
+import * as express from "express";
+import * as expressBunyan from "express-bunyan-logger";
+import * as passport from "passport";
+import { ITokenPayload, BearerStrategy, VerifyCallback, IBearerStrategyOption } from "passport-azure-ad";
+import { signalRouterCreator, IPeerRequest, PeerList } from "webrtc-signal-http-ts";
+import { ISignalerOpts, IPeerBearerStrategyOpts, IExpressApp } from "./modules";
+const heartbeatRouterCreator = require("webrtc-signal-http-heartbeat");
```

Comment:
We should be consistent, either use the `require` pattern or `import`. Since we moved to ts, we should change all to `import`

Reply:
To import them we need an import of @Types/module which does not exist for these packages. I tried to make a universal import, but this error prevented it. I can manually add an index.d.ts file to node_modules, but I figured it was not a long term solution and as they were ported, we could switch the imports

Reply:
You only need to create a new file `modules.d.ts` at the root of the project, not inside node_modules. You then declare like this `declare module 'webrtc-signal-http-heartbeat';` You can do this for every import that you have. After you have the modules file, you can then import like this `import * as heartbeatRouterCreator  from "webrtc-signal-http-heartbeat";` and the error will go away. 

You might also need to add the `modules.d.ts` file in `tsconfig.json` under `include`

Reply:
done! easy fix. thank you!

Reply:
@bengreenier including you here for context